src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/DefaultRazorSemanticTokenInfoService.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Completion;
+using Microsoft.VisualStudio.Editor.Razor;
+using SyntaxNode = Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
+{
+    internal class DefaultRazorSemanticTokenInfoService : RazorSemanticTokenInfoService
+    {
+        private readonly SemanticTokenLegend _symanticTokenLegend;
+
+        [ImportingConstructor]
+        public DefaultRazorSemanticTokenInfoService()
+        {
+            _symanticTokenLegend = new SemanticTokenLegend();
+        }
+
+        public override SemanticTokens GetSemanticTokens(RazorCodeDocument codeDocument, SourceLocation? location = null)
+        {
+            if (codeDocument is null)
+            {
+                throw new ArgumentNullException(nameof(codeDocument));
+            }
+            var syntaxTree = codeDocument.GetSyntaxTree();
+
+            var syntaxTokens = VisitAllNodes(syntaxTree);
```

Comment:
Are all the returned nodes `SyntaxToken`s? If so, can we have this return `SyntaxToken` instead of `SyntaxNode`?

Reply:
Looking more, I realized that none of these would be `SyntaxToken`s. Suggest renaming this to `syntaxTokenNodes` for clarity.